Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -398,16 +414,13 @@ void PredictPeaks::setStructureFactorCalculatorFromSample(
  * @param orientedUB
  * @param goniometerMatrix
  */
-void PredictPeaks::calculateQAndAddToOutput(V3D &hkl,
+void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
                                             const DblMatrix &orientedUB,
                                             const DblMatrix &goniometerMatrix) {
   // The q-vector direction of the peak is = goniometer * ub * hkl_vector
   // This is in inelastic convention: momentum transfer of the LATTICE!
   // Also, q does have a 2pi factor = it is equal to 2pi/wavelength.
-  if (convention == "Crystallography") {
-    hkl = hkl * (-1.0);
-  }
```

Comment:
By changing hkl to a const and not changing the sign, the output of PredictPeaks has the wrong sign of hkl when using Crystallography Convention.  Fixing this in #18149.  Please ask me or Andrei to check any changes to Crystallography Convention.